fdbclient/SpecialKeySpace.actor.h

```diff
@@ -33,27 +33,57 @@
 #include "fdbclient/ReadYourWrites.h"
 #include "flow/actorcompiler.h" // This must be the last #include.
 
-class SpecialKeyRangeBaseImpl {
+class SpecialKeyRangeReadImpl {
 public:
 	// Each derived class only needs to implement this simple version of getRange
 	virtual Future<Standalone<RangeResultRef>> getRange(ReadYourWritesTransaction* ryw, KeyRangeRef kr) const = 0;
 
-	explicit SpecialKeyRangeBaseImpl(KeyRangeRef kr) : range(kr) {}
+	explicit SpecialKeyRangeReadImpl(KeyRangeRef kr) : range(kr) {}
 	KeyRangeRef getKeyRange() const { return range; }
 	// true if the getRange call can emit more than one rpc calls,
 	// we cache the results to keep consistency in the same getrange lifetime
 	// TODO : give this function a more descriptive name
 	virtual bool isAsync() const { return false; }
 
-	virtual ~SpecialKeyRangeBaseImpl() {}
+	virtual ~SpecialKeyRangeReadImpl() {}
 
 protected:
 	KeyRange range; // underlying key range for this function
 };
 
-class SpecialKeyRangeAsyncImpl : public SpecialKeyRangeBaseImpl {
+class ManagementAPIError {
 public:
-	explicit SpecialKeyRangeAsyncImpl(KeyRangeRef kr) : SpecialKeyRangeBaseImpl(kr) {}
+	static std::string toJsonString(bool retriable, std::string command, std::string msg) {
+		json_spirit::mObject errorObj;
```

Comment:
We should add some schema validation, and document the schema

Reply:
Thanks. I have added the tests for the schema check.
Documentation I will add in the following pr.